app/multitenant/aws_collector.go

```diff
@@ -391,24 +409,27 @@ func (c *awsCollector) Add(ctx context.Context, rep report.Report, buf []byte) e
 
 	var resp *dynamodb.PutItemOutput
 	err = instrument.TimeRequestHistogram(ctx, "DynamoDB.PutItem", dynamoRequestDuration, func(_ context.Context) error {
-		var err error
-		resp, err = c.db.PutItem(&dynamodb.PutItemInput{
-			TableName: aws.String(c.tableName),
-			Item: map[string]*dynamodb.AttributeValue{
-				hourField: {
-					S: aws.String(rowKey),
-				},
-				tsField: {
-					N: aws.String(colKey),
-				},
-				reportField: {
-					S: aws.String(reportKey),
+		return withBackoff(func() error {
+			var err error
+			resp, err = c.db.PutItem(&dynamodb.PutItemInput{
+				TableName: aws.String(c.tableName),
+				Item: map[string]*dynamodb.AttributeValue{
+					hourField: {
+						S: aws.String(rowKey),
+					},
+					tsField: {
+						N: aws.String(colKey),
+					},
+					reportField: {
+						S: aws.String(reportKey),
+					},
 				},
-			},
-			ReturnConsumedCapacity: aws.String(dynamodb.ReturnConsumedCapacityTotal),
+				ReturnConsumedCapacity: aws.String(dynamodb.ReturnConsumedCapacityTotal),
+			})
+			return err
```

Comment:
Surely we should only back off on `provisionedThroughputExceededException`. That's what the [cortex code](https://github.com/weaveworks/cortex/blob/611bf7c1561a0318a9303e7b0683ac433b03c870/chunk/dynamodb_client.go#L98) does.